docs/CPS/CPS-XXXX/README.md

```diff
@@ -0,0 +1,129 @@
+
+
+> [!NOTE]
+> Requests are also made to fetch data on-demand but the scope of this document is to focus on the fundamental wallet requirements, so it's not mentioned here.
+
+<img alt="Lace fundamental state sequence flow diagram" src="images/Lace_Fundamental_State_Sequence_Flow.png">
+
+#### Why?
+- The philosophy in Lace is to place the responsibility of deriving wallet state from the historical transactions, similar to projecting state via the chain-sync protocol.
+- Blockfrost was used during the early development of the Lace frontend, however the design of the API to require many small requests conflicts with the rollback handling strategy of Lace, and therefore was not a good fit for fetching transactions. 
+- A custom API was implemented for this data access, with a batch interface, to query the `cardano-db-sync` database, and later custom projections with more a more optimised schema.
+- The API schema was implemented from scratch, based on the domain model, due to the lack of an existing standard.
+
+> [!NOTE]
+> Since this API was developed, [Cardano DB Sync] has since added a mode to store transactions in CBOR, but at the time of implementation it was not an option.
+
+#### Current Alternatives
+[Blockfrost WebSocket Link] is a partial solution for the impedance mismatch problem, however it's not addressing the fragility and standards problem. It's not practical to run this service to satisfy a single-tenant, full node, use case. 
+
+## Goals
+- Define a standard API that is optimal for fundamental wallet operation. Consider extensibility for optional enhancement.
+- Deliver data as a filtered subset of the network state, using native encoding formats, to simplify server implementations and push processing to the edge of the network.
+- Foster long term stability of the API when crossing major Cardano protocol upgrade boundaries.
+- Facilitate event-driven wallet behavior, minimising the need for polling, and to increase the sophistication of infrastructure management and user-base insights.
+- Support script-based wallets
+- Minimise the network demand by allowing the server to control the throttling strategy, rather than a pull-based approach like we see in the local chain-sync protocol. 
+- API can be applied in a local, single-tenant environment, and a remote multi-tenant deployment, to facilitate a universal approach to wallet supporting an optional full local node for power users. 
+- Remove or reduce the need for wallets to perform address discovery
+
+## Open Questions
```

Comment:
There are two more open questions in my mind. I am not sure if they belong here. My comments under the questions are to make the questions more transparent.
- How can we keep re-connections cheap for the client and the server?
     In the case of mobile devices, I suspect there will be plenty of one-second disconnections in a fast-moving car.
- How can we ensure that the first sync of a big wallet is time and network-efficient?
     Considering a busy address, pushing relevant events one by one from the genesis might be time and network-heavy. Is a hybrid solution of sending a snapshot acceptable? Is sending the messages as a batch acceptable?

Reply:
https://github.com/will-break-it/wallet-architecture/pull/3/commits/80b1097e379cc43b58f39b1086386d97f79205f5